security_monkey/watchers/s3.py

```diff
@@ -58,10 +59,19 @@ def slurp_items(**kwargs):
 
             for bucket_name in bucket_names:
                 bucket = self.process_bucket(bucket_name, name=bucket_name, **kwargs)
-                if bucket:
+
+                # Need to check accessing the bucket was successful:
+                if bucket and not bucket.get("Error"):
                     item = S3Item.from_slurp(bucket_name, bucket, **kwargs)
                     item_list.append(item)
 
+                # If there was an error, add it to the exception map:
+                if bucket.get("Error"):
```

Comment:
Is it cleaner to pull the exception map and fill it directly, or to just throw the exception and let the iter_account_region decorator handle it?  Either way works.